### PR TITLE
[Merged by Bors] - exclude active set from ballot signature

### PR DESCRIPTION
**Pull request by countvonzero**
## Motivation
first part of #4237
part of #4087

## Changes
- exclude active set from ref ballot signature. include the active set instead.
- add a EligibilityCount in ballot (for total number of ballots in the epoch) to prepare for future atx pruning

### CODE REVIEW

**Comment by countvonzero** [comment]
bors try

File: common/types/ballot.go [hunk 1]
```diff
@@ -101,6 +105,9 @@ type InnerBallot struct {
 	// can be verified before decoding votes.
 	OpinionHash Hash32
 
+	// total number of ballots the smesher is eligible in this epoch.
+	EligibilityCount uint32
```
Comment (dshulyak) [inline common/types/ballot.go:109]: we need to validate that this number matches computed number, or do the alternative. easiest would be to validate it at (*Validator).CheckEligibility

if you don't validate it, and later this number is used in the tortoise - ballot weight will be incorrect. for example computed eligibility count is 10 and this was used when ballots were accepted in Validator.CheckEligibility, node accepted 10 ballots this way. but if EligibilityCount is 1 and it is used in tortoise, then every ballot will have weight of the full atx weight

alternative is to validate only ballots from the current epoch using computed count, and for previous epoch use specified count in the oracle

**Review by dshulyak — APPROVED** [review]
please fix validation stuff (in case i didn't miss it)

File: common/types/ballot.go (hunk 1 above)
Comment (countvonzero) [reply common/types/ballot.go:109]: thanks. i missed that. added validation as part of (*Validator).CheckEligibility

**Comment by countvonzero** [comment]
bors try

**Comment by countvonzero** [comment]
bors merge

**Pull request closed without merging**